os/mm/kmm_heap/kmm_realloc.c

```diff
@@ -158,7 +166,7 @@ FAR void *kmm_realloc(FAR void *oldmem, size_t newsize)
 		}
 	}
 
-	mm_manage_alloc_fail(kheap_new, CONFIG_KMM_NHEAPS, newsize, KERNEL_HEAP);
+	mm_manage_alloc_fail(kheap_new, 0, CONFIG_KMM_NHEAPS, newsize, KERNEL_HEAP);
```

Comment:
All of allocation function use the same value for the start and the end.
The start is 0, the end is CONFIG_KMM_NHEAPS.
How about using definitions for them? You can use them in two locations of each file.
This way gives
1. good readability
2. prevent mismatch